Bottom: 9975ad326d9c221542d4ac04a4d8afdfc544a331
Top:    50b804f3123beb31632cfbe4a3ddee31e9dec1f3
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-12-25 02:14:25 +0000

Try a new, different sized VBO buffer for each upload.


---

diff --git a/src/hid/common/hidgl.c b/src/hid/common/hidgl.c
index f0ded25..b037a7a 100644
--- a/src/hid/common/hidgl.c
+++ b/src/hid/common/hidgl.c
@@ -135,10 +135,6 @@ hidgl_init_triangle_array (triangle_buffer *buffer)
    */
   buffer->use_map = false;
 
-  /* If using VBOs (but not mapping), we only need to this once */
-  if (buffer->use_vbo && !buffer->use_map)
-    glBufferData (GL_ARRAY_BUFFER, BUFFER_SIZE, NULL, GL_STREAM_DRAW);
-
   buffer->triangle_array = NULL;
   hidgl_reset_triangle_array (buffer);
 }
@@ -176,10 +172,10 @@ hidgl_flush_triangles (triangle_buffer *buffer)
       glUnmapBuffer (GL_ARRAY_BUFFER);
       buffer->triangle_array = NULL;
     } else {
-      /* NB: We only upload the portion of the buffer we've used */
-      glBufferSubData (GL_ARRAY_BUFFER, 0,
-                       BUFFER_STRIDE * buffer->vertex_count,
-                       buffer->triangle_array);
+      glBufferData (GL_ARRAY_BUFFER,
+                    BUFFER_STRIDE * buffer->vertex_count,
+                    buffer->triangle_array,
+                    GL_STREAM_DRAW);
     }
   } else {
     data_pointer = buffer->triangle_array;
